doc/source/whatsnew/v2.0.0.rst

```diff
@@ -238,6 +238,9 @@ Copy-on-Write improvements
   a modification to the data happens) when constructing a Series from an existing
   Series with the default of ``copy=False`` (:issue:`50471`)
 
+- The :class:`DataFrame` constructor now will keep track of references when called
+  with another :class:`DataFrame` or ``BlockManager``.
```

Comment:
Not sure if mentioning BlockManager here is worth it?

Reply:
No, I wouldn't mention that (normal users should never pass that)

Reply:
Thx, removed